### ISSUE TITLE
Freehand shape can permanently break map geometry

### ISSUE THREAD

**Issue by mcglincy** [mcglincy/dungeondraw-foundry-vtt#76]
https://discord.com/channels/170995199584108546/722566083499786352/930643277441536021

Looks like somehow the non-linestring breaking shape can get into the geometry, and then break any unions afterward.

```
Uncaught (in promise) TopologyException: found non-noded intersection between LINESTRING ( 1950 1400, 2128 1522 ) and LINESTRING ( 1916 1618, 2266 1336 ) [ (2077.828281088529, 1487.6126420943851, NaN) ]
    at oi.checkValid (http://localhost:30000/modules/dungeon-draw/modules/dungeondraw-bundle.min.js:1:260421)
    at ai.checkValid (http://localhost:30000/modules/dungeon-draw/modules/dungeondraw-bundle.min.js:1:261069)
    at Function.checkValid (http://localhost:30000/modules/dungeon-draw/modules/dungeondraw-bundle.min.js:1:261034)
    at Fi.computeOverlay (http://localhost:30000/modules/dungeon-draw/modules/dungeondraw-bundle.min.js:1:296838)
    at Fi.getResultGeometry (http://localhost:30000/modules/dungeon-draw/modules/dungeondraw-bundle.min.js:1:296325)
    at Function.overlayOp (http://localhost:30000/modules/dungeon-draw/modules/dungeondraw-bundle.min.js:1:292642)
    at Ci.getResultGeometry (http://localhost:30000/modules/dungeon-draw/modules/dungeondraw-bundle.min.js:1:275236)
    at Function.overlayOp (http://localhost:30000/modules/dungeon-draw/modules/dungeondraw-bundle.min.js:1:274914)
    at Function.union (http://localhost:30000/modules/dungeon-draw/modules/dungeondraw-bundle.min.js:1:313767)
    at fr (http://localhost:30000/modules/dungeon-draw/modules/dungeondraw-bundle.min.js:1:327165)
```

**Comment by mcglincy** [comment]
Here's an example shape that seems to break everything after it (sort of a sloppy circle with extended line segments):

![image](https://user-images.githubusercontent.com/189172/149149381-6d73d6c4-a57d-4b93-a504-ea7fbc540e93.png)

**Comment by mcglincy** [comment]
Added isValid() check and ui error notification. Fixed in 0.17.0.
